### DISCUSSION TITLE
Delete page

### DISCUSSION THREAD

**Discussion by chibyk101** [filamentphp/filament#2288] [Help] [answered]
### Package

filament/filament

### Package Version

v2.0.0

### Laravel Version

v9.2

### PHP Version

v8.0.18

### Bug description

After deleting pages I get a "route not registered error from laravel" I followed the documentation which says

"If you'd like to delete a page from your resource, you can just delete the page file from the Pages directory of your resource, and its entry in the getPages() method.

For example, you may have a resource with records that may not be created by anyone. Delete the Create page, and then remove it from getPages():"

but somehow the views still make reference to the default routes (index, create, edit)

### Steps to reproduce

create a resource. Eg. WebsiteSettingResource
create your own custom view "php artisan make:filament-page WebsiteSetup --resource=WebsiteSettingResource"
Remove registered pages from the getPages() method
delete pages from the WebsiteSettingResource\Pages directory

There you go!
Error: Route [filament.resources.websites.index] not defined.

**Comment by danharrin** [comment, marked as answer]
you need to keep the index page at least... otherwise what is the point of the resource?

**Reply by Z3d0X** [reply]
You can remove index page too... if you disable navigation registration

```php
protected static bool $shouldRegisterNavigation = false;
```

**Reply by Z3d0X** [reply]
> otherwise what is the point of the resource?

😅 I have empty resources, just so `filament-shield` registers them and generates policy & permissions for them.

PS: they are actually used in RelationManagers
```php
class VlanResource extends Resource
{
    protected static ?string $model = Vlan::class;
    protected static bool $shouldRegisterNavigation = false;
}
```

**Reply by chibyk101** [reply]
I will try this out thank you
